File: app/src/main/java/com/derlados/computerconf/Internet/GsonSerializers/HashMapDeserializer.java
package com.derlados.computerconf.Internet.GsonSerializers;

import com.google.gson.Gson;
import com.google.gson.JsonDeserializationContext;
import com.google.gson.JsonDeserializer;
import com.google.gson.JsonElement;
import com.google.gson.JsonParseException;
import com.google.gson.reflect.TypeToken;

import java.lang.reflect.Type;
import java.util.HashMap;

public class HashMapDeserializer implements JsonDeserializer<HashMap<String, String>> {
    @Override
    public HashMap<String, String> deserialize(JsonElement json, Type typeOfT, JsonDeserializationContext context) throws JsonParseException {

        String jsonString = json.getAsJsonArray().get(0).toString();
        HashMap<String, String> map = new HashMap<>();
        Type type = new TypeToken<HashMap<String, String>>() {}.getType();
        map = new Gson().fromJson(jsonString, type);

        return map;
    }
}

File: app/src/main/java/com/derlados/computer_conf/internet/gson_serializers/HashMapDeserializer.kt
package com.derlados.computer_conf.internet.gson_serializers

import android.os.Debug
import android.util.Log
import com.google.gson.*
import com.google.gson.reflect.TypeToken
import java.lang.reflect.Type
import java.util.*
import kotlin.Throws

class HashMapDeserializer : JsonDeserializer<HashMap<String?, String?>> {
    @Throws(JsonParseException::class)
    override fun deserialize(json: JsonElement, typeOfT: Type, context: JsonDeserializationContext): HashMap<String?, String?> {
       Log.d(Log.DEBUG.toString(), json.toString())

        val jsonString = json.asJsonArray[0].toString()
        var map: HashMap<String?, String?>
        val type = object : TypeToken<HashMap<String?, String?>?>() {}.type
        map = Gson().fromJson(jsonString, type)
        return map
    }


}